unikernel.ml

```diff
@@ -1093,24 +1157,23 @@ struct
                 authenticate !store reqd (dashboard !albatross reqd))
         | "/admin/users" ->
             check_meth `GET (fun () ->
-                (* TODO: a middleware for admins *)
-                authenticate !store reqd (users !store reqd))
+                authenticate ~check_admin:true !store reqd (users !store reqd))
         | "/admin/settings" ->
             check_meth `GET (fun () ->
-                (* TODO: a middleware for admins *)
-                authenticate !store reqd (settings !store reqd))
+                authenticate ~check_admin:true !store reqd
+                  (settings !store reqd))
         | "/api/admin/settings/update" ->
             check_meth `POST (fun () ->
-                (* TODO: a middleware for admins *)
-                authenticate !store reqd
+                authenticate ~check_admin:true ~api_meth:true !store reqd
                   (update_settings stack store albatross reqd))
         | "/api/admin/user/status/toggle" ->
             check_meth `POST (fun () ->
-                (* TODO: a middleware for admins *)
-                authenticate !store reqd (toggle_user store reqd))
-        | "/unikernel-info" ->
+                authenticate ~check_admin:true ~api_meth:true !store reqd
+                  (toggle_user store reqd))
+        | "/api/admin/unikernels" ->
             check_meth `GET (fun () ->
-                authenticate !store reqd (unikernel_info !albatross reqd))
+                authenticate ~check_admin:true ~api_meth:true !store reqd
+                  (unikernel_info !albatross reqd))
```

Comment:
why is this `unikernel_info` now requiring admin privileges?

Reply:
This endpoint is actually not being used, but I give it the admin privileges as my plan is that it should be an endpoint that list all unikernels from all users.

Reply:
I'm not sure, can we defer that to a later time (taking https://github.com/robur-coop/mollymawk/issues/38 into account, which is about the REST API)?